### PR TITLE
Improve postfix trace

### PR DESCRIPTION
**Pull request by RblSb**
`call().tr<tab>;` currently generates `trace(call());;`, so this fixes such case with semicolon

### CODE REVIEW

File: src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx [hunk 1]
```diff
@@ -139,6 +146,16 @@ class PostfixCompletion {
 		return result;
 	}
 
+	function createTraceItem(expr:String, afterExprChar:String, add:PostfixCompletionItem->Void):Void {
+		final endChar = afterExprChar == "\n" ? ";" : "";
```
Comment (AlexHaxe) [inline src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx:150]: shouldn't it be the other way around, e.g. add a semicolon if `afterExprChar` is not a semicolon? (not sure if looking just for `\n` works on Windows) 
and since `trace` doesn't really return anything there is no need to omit that semicolon in most cases. or more precise the only case where you don't want to emit a semicolon is when one is already there.

technically we would have to read more than one character for `afterExprChar` but I think users can live with  that - the usecase of having a stray semicolon when doing postfix completion is probably already exotic enough that we don't have to scan multiple characters past current edit-point.

Comment (RblSb) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx:150]: That was my initial idea, but this works better for corner cases like `[].iter(item -> item.tr<tab>)` and maybe some other. I tested on Windows and it is fine in default LF mode, but it should test for "\n" or "\r", yeah.

Comment (kLabz) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx:150]: Is it really that useful to add `;` in the first place?

Seems a bit out of scope for such snippet to me, and is usually the kind of "thought you'd like me to also do X" things that can (imo) be annoying (especially with oversight like that double `;` issue :sweat_smile:)

Comment (AlexHaxe) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx:150]: I'm also fine with no semicolon at all. the only problem with that is, that when you complete `xyz.tr|` to `trace(xyz)` your instinct is to type a semicolon. only the cursor still has `xyz` marked, so you will replace `xyz` with a semicolon, instead of adding one at the end. 
it's a bit weird and I must admit I've never used `trace` postfix completion so maybe it takes some getting used to. the only time I use postfix completion is with `switch`.

Comment (RblSb) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: I don't want to mess with people's muscle memory and think that adding `;` can be left if no more problems arise after this fix. After all, trace does not return values ​​and almost always requires a semicolon at the end anyway.

Comment (kLabz) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: I don't think that's enough

See these for example:
```haxe
var foo = {bar: () -> trace("foo")};
var bar = [() -> trace("foo")]
```

And variations with whitespace, _including_ new lines.

Comment (RblSb) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: Instead of solving a lot of rare cases, I noticed the implementation of eating and improved it a bit for trace/var/final snippets. I think that's good enough for their main usage

Comment (kLabz) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: How exactly does that fix any of these cases?

Comment (RblSb) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: It does not. I decided to fix the main cases `trace(0);; / final a = 0;; / var a = 0;;`, the rest will require separate additions. But I don't think they are as important in the real world, as well as my example with `[].iter(...)`.

Comment (kLabz) [reply src/haxeLanguageServer/features/haxe/completion/PostfixCompletion.hx]: The fix was not to add semicolon at all, instead of adding it that only works in simple cases :/
